File: src/main/java/kitchenpos/menus/infra/ProductClientImpl.java
package kitchenpos.menus.infra;

import kitchenpos.menus.tobe.domain.menu.ProductClient;
import kitchenpos.products.tobe.domain.Product;
import kitchenpos.products.tobe.domain.ProductRepository;
import org.springframework.stereotype.Service;

import java.math.BigDecimal;
import java.util.List;
import java.util.UUID;

@Service
public class ProductClientImpl implements ProductClient {
    private final ProductRepository productRepository;

    public ProductClientImpl(ProductRepository productRepository) {
        this.productRepository = productRepository;
    }

    @Override
    public List<Product> findAllByIdIn(List<UUID> productIds) {
        return productRepository.findAllByIdIn(productIds);
    }

    @Override
    public BigDecimal getProductPrice(UUID productId) {
        return productRepository.findById(productId).orElseThrow(IllegalArgumentException::new).getPrice();
    }
}

File: src/main/kotlin/kitchenpos/tobe/menu/infra/ProductClientImpl.kt
package kitchenpos.tobe.menu.infra

import kitchenpos.tobe.menu.domain.ProductQueryClient
import kitchenpos.tobe.menu.domain.dto.ProductPrice
import kitchenpos.tobe.product.domain.repository.ProductRepositoryV2
import org.springframework.stereotype.Service
import java.util.*

@Service
class ProductClientImpl(
    private val productRepository: ProductRepositoryV2,
) : ProductQueryClient {
    override fun getProductPrices(productIds: List<UUID>): List<ProductPrice> {
        return productRepository.findAllByIdIn(productIds).map {
            ProductPrice(
                id = it.id,
                price = it.getPrice(),
            )
        }
    }
}
